src/ResourceManager/Relay/Commands.Relay/Cmdlets/AuthorizationRule/SetAzureRelayAuthorizationRules.cs

```diff
@@ -0,0 +1,128 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Relay.Models;
+using Microsoft.Azure.Management.Relay.Models;
+using System.Management.Automation;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.Relay.Commands
+{
+    /// <summary>
+    /// 'Set-AzureRmRelayAuthorizationRule' Cmdlet updates the specified AuthorizationRule
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, RelayAuthorizationRuleVerb, DefaultParameterSetName = NamespaceAuthoRuleParameterSet, SupportsShouldProcess = true), OutputType(typeof(AuthorizationRuleAttributes))]
+    public class SetAzureRelayAuthorizationRule : AzureRelayCmdletBase
+    {
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
```

Comment:
Scenario for right :  Rights (properties) will be only if Inputobject is not used.
The scenarios will be as follows : 
-RGName, -Namespace, -Name, -InputObject/-Rights
-RGName, -Namespace, -WcfRelay/-HybirdConnection, -Name,  -InputObject/-Rights. 

please let me know  your views.